src/sqlfluff/rules/L044.py

```diff
@@ -120,7 +120,7 @@ def _analyze_result_columns(self, query: Query):
                         if isinstance(o, Query):
                             self._analyze_result_columns(o)
                             return
-                    assert False, "Should be unreachable"  # pragma: no cover
+                    raise RuleFailure(query.selectables[0].selectable)
```

Comment:
Do we want an accompanying `self.logger.debug(` message here?

What does `raise RuleFailure(query.selectables[0].selectable)` look like to us if we were trying to debug this?

Reply:
This is something specific to this rule -- this is how it reports lint issues. The rule itself catches this and returns a regular LintResult. (You'll see a very similar line 10 lines up in this file.) I designed it this way because this rule recourses deeply into the parse structure, and it seemed cleaner to raise an exception rather than returning the result step by step up the call stack the usual way.

Reply:
For this reason, I don't think I log message is needed.

Reply:
makes sense, the example 10 lines up has the log message before which was why I suggested it 😛

Happy to approve if you're happy it's sufficient 👍

Reply:
Ah, good point. I think some days I feel more "loggy" than others. I'll go ahead and add that, just a sec.

Reply:
Log message added. 👍